test/json_schemer_test.rb

```diff
@@ -354,6 +354,141 @@ def test_it_returns_correct_pointers_for_ref_id
     }
   end
 
+  def test_it_returns_correct_pointers_for_a_nested_ref_id
+    root = {
+     'foo' => {
+        'bar' => {
+          '$id' => '#bar',
+          'type' => 'string'
+        }
+      },
+      'properties' => {
+        'a' => {
+          'properties' => {
+            'x' => { '$ref' => '#bar' }
+          }
+        }
```

Comment:
```suggestion
      '$ref' => '#bar'
```

This makes the test shorter and avoids confusion about what "nested" in the test title refers to.